app/src/main/java/com/willowtree/vocable/splash/SplashViewModel.kt

```diff
@@ -0,0 +1,159 @@
+package com.willowtree.vocable.splash
+
+import android.content.Context
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.squareup.moshi.Moshi
+import com.willowtree.vocable.BaseViewModel
+import com.willowtree.vocable.R
+import com.willowtree.vocable.presets.PresetsRepository
+import com.willowtree.vocable.room.Category
+import com.willowtree.vocable.room.CategoryPhraseCrossRef
+import com.willowtree.vocable.room.Phrase
+import com.willowtree.vocable.room.PresetsObject
+import com.willowtree.vocable.utils.VocableSharedPreferences
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.koin.core.get
+import org.koin.core.inject
+import java.nio.charset.Charset
+import java.util.*
+
+class SplashViewModel : BaseViewModel() {
+
+    private val presetsRepository: PresetsRepository by inject()
+    private val moshi: Moshi by inject()
+    private val sharedPrefs: VocableSharedPreferences by inject()
+    private val numbersCategoryId = get<Context>().getString(R.string.category_123_id)
+    private val mySayingsCategoryId = get<Context>().getString(R.string.category_my_sayings_id)
+
+    private val liveExitSplash = MutableLiveData<Boolean>()
+    val exitSplash: LiveData<Boolean> = liveExitSplash
+
+    init {
+        populateDatabase()
+    }
+
+    private fun populateDatabase() {
```

Comment:
I agree with Daniel, this ties the database initiation closely to the splash viewmodel when ideally the splash viewmodel should just tell the repository to do it's thing and let it know when it's done so it can proceed.

Reply:
Moved this logic into the repository!